src/backend/base/langflow/components/tools/calculator.py

```diff
@@ -41,20 +41,15 @@ def build_tool(self) -> Tool:
         )
 
     def _eval_expr(self, node):
-        # Define the allowed operators
-        operators = {
-            ast.Add: operator.add,
-            ast.Sub: operator.sub,
-            ast.Mult: operator.mul,
-            ast.Div: operator.truediv,
-            ast.Pow: operator.pow,
-        }
         if isinstance(node, ast.Num):
             return node.n
         if isinstance(node, ast.BinOp):
-            return operators[type(node.op)](self._eval_expr(node.left), self._eval_expr(node.right))
+            left_val = self._eval_expr(node.left)
+            right_val = self._eval_expr(node.right)
+            return self.operators[type(node.op)](left_val, right_val)
```

Comment:
Just by curiosity: does going through an intermediate variable help with performance ?

Reply:
not really, it should be a no-op wrt performance. I believe the llm thought this might look more readable?